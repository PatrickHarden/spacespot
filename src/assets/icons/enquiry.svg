<?xml version="1.0" encoding="UTF-8"?>
<svg width="30px" height="28px" viewBox="0 0 30 28" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <!-- Generator: sketchtool 63 (101010) - https://sketch.com -->
    <title>F65E5ADF-220D-4D3F-9779-B9713793BF78</title>
    <desc>Created with sketchtool.</desc>
    <g id="Header" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
        <g id="Full-home-page-expanded-Copy-3" transform="translate(-292.000000, -748.000000)" fill="#FFFFFF" fill-rule="nonzero">
            <g id="enquiry-icon" transform="translate(292.000000, 748.000000)">
                <path d="M26.4983523,0.262180398 L9.76683239,0.22787642 C8.16434659,0.22787642 6.86079545,1.53142756 6.86079545,3.13391335 L6.86079545,5.64496449 L2.90603693,5.65329545 C1.30355114,5.65329545 0,6.95684659 0,8.55933239 L0,18.9196236 C0,20.5221094 1.30355114,21.8251705 2.90603693,21.8251705 L5.88068182,21.8251705 L5.88068182,26.7257386 C5.88068182,26.9281321 6.00515625,27.1099432 6.19431818,27.1829616 C6.25165483,27.2050142 6.31144176,27.2157955 6.37073864,27.2157955 C6.50599432,27.2157955 6.63879972,27.1594389 6.73338068,27.0555469 L11.488892,21.8241903 L19.6365767,21.7908665 C21.2390625,21.7908665 22.5426136,20.4873153 22.5426136,18.8853196 L22.5426136,18.8299432 L25.1203125,21.6654119 C25.2148935,21.769304 25.3476989,21.8256605 25.4829545,21.8256605 C25.5422514,21.8256605 25.6020384,21.8148793 25.659375,21.7928267 C25.8485369,21.7198082 25.9730114,21.5379972 25.9730114,21.3356037 L25.9730114,16.4350355 L26.4973722,16.4350355 C28.099858,16.4350355 29.4034091,15.1314844 29.4034091,13.5294886 L29.4034091,3.16821733 C29.4034091,1.56622159 28.100348,0.262670455 26.4983523,0.262180398 Z M21.5625,18.8848295 C21.5625,19.9467827 20.6985298,20.8102628 19.6346165,20.8102628 L11.2693466,20.8445668 C11.1321307,20.8450568 11.0007955,20.9033736 10.9086648,21.0048153 L6.86079545,25.4579616 L6.86079545,21.3346236 C6.86079545,21.0641122 6.64174006,20.8445668 6.37073864,20.8445668 L2.90603693,20.8445668 C1.84408381,20.8445668 0.980113636,19.9805966 0.980113636,18.9191335 L0.980113636,8.55884233 C0.980113636,7.4968892 1.84408381,6.63291903 2.90701705,6.63291903 L7.35085227,6.62360795 C7.35085227,6.62360795 7.35085227,6.62360795 7.35085227,6.62360795 C7.35134233,6.62360795 7.35134233,6.62360795 7.35183239,6.62360795 L19.6365767,6.598125 C20.6985298,6.598125 21.5625,7.46209517 21.5625,8.5240483 L21.5625,17.5611861 L21.5625,18.8848295 Z M28.4232955,13.5285085 C28.4232955,14.5904616 27.5593253,15.4539418 26.4973722,15.4539418 L25.4829545,15.4539418 C25.2119531,15.4539418 24.9928977,15.6734872 24.9928977,15.9439986 L24.9928977,20.0673366 L22.5426136,17.3720241 L22.5426136,8.52453835 C22.5426136,6.92205256 21.2390625,5.61850142 19.6355966,5.61850142 L7.84090909,5.6425142 L7.84090909,3.1334233 C7.84090909,2.07147017 8.70487926,1.2075 9.76585227,1.2075 L26.4968821,1.24180398 C26.4973722,1.24180398 26.4973722,1.24180398 26.4978622,1.24180398 C27.5598153,1.24180398 28.4237855,2.10577415 28.4237855,3.16772727 L28.4237855,13.5285085 L28.4232955,13.5285085 Z" id="Shape"></path>
            </g>
        </g>
    </g>
</svg>