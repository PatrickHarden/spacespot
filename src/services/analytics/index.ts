import ReactGA from 'react-ga'
import axios from 'axios'

import config from 'config'
import history from 'browserhistory'
import { loadPixelAPI } from './loadpixel'
export { usePixel } from './hooks'

let ready = false

const init = () => {
  const consent = localStorage.getItem('cookie-consent')
  if (!consent) return
  ReactGA.initialize(config.GOOGLE_TRACKING_CODE)
  ready = true
  // console.log('Analytics: init')

  // Initialize google analytics page view tracking
  history.listen(location => {
    // console.log('Analytics: pageView ', location.pathname)
    ReactGA.set({ page: location.pathname }) // Update the user's current page
    ReactGA.pageview(location.pathname) // Record a pageview for the given page
  })

  if (consent !== 'false') {
    loadPixelAPI()
  }
}

const event = (ev: any) => {
  if (!ready) return
  // console.log('Analytics: event', ev)
  ReactGA.event(ev)
}

/**
 * GTM dataLayer push
 *
 * see https://developers.google.com/tag-manager/devguide#events
 */
const gtmEvent = (ev: any) => {
  const win = (window as unknown) as any
  win.dataLayer && win.dataLayer.push(ev)
}

const pixel = (ct: string) => {
  const consent = localStorage.getItem('cookie-consent')

  if (!ready) return
  try {
    if (consent !== 'false') {
      axios.get(
        // Aded withCredentials to forward the token for Campaign
        //`https://insight.adsrvr.org/track/pxl/?adv=6gmmyim&ct=${ct}&fmt=3`,
        `https://insight.adsrvr.org/track/pxl/?adv=6gmmyim&ct=${ct}&fmt=3`, { withCredentials: true },
      )
    }
  } catch {
    // ignore pixel issues
  }
}

const PIXELS = {
  SEARCH: '0:ka4bzg6',
  SELECT_SPACE: '0:btb7pw8',
  SEND_ENQUIRY: '0:6ywcv06',
  SIGNUP: '0:6u9gs4z',
  ACCEPT_ENQUIRY: '0:9jr4h2a',
  NEGOTIATION_SUBMITTED: '0:pcdfo63',
  ACCEPT_LEASE: '0:2tge50w',
  SIGN: '0:bqnev20',
}

export default {
  init,
  event,
  gtmEvent,
  pixel,
  PIXELS,
}
