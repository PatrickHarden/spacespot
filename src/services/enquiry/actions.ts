import {
  EnquiryAction,
  EnquiryResponse,
  DocumentDTO,
  DocData,
  EnquiryRequest,
  EnquiryRequestFlex,
} from './types'
import { FileItem } from 'services/global/types'
import constants from './constants'

const init = (payload: EnquiryRequest | EnquiryRequestFlex) => ({
  type: constants.INIT,
  payload,
})

const initSuccess = (payload: EnquiryResponse): EnquiryAction => ({
  type: constants.INIT_SUCCESS,
  payload,
})

const initError = (payload: string): EnquiryAction => ({
  type: constants.INIT_ERROR,
  payload,
})

const get = (payload: string): EnquiryAction => ({
  type: constants.GET,
  payload,
})

const getSuccess = (payload: EnquiryResponse): EnquiryAction => ({
  type: constants.GET_SUCCESS,
  payload,
})

const getError = (payload: string): EnquiryAction => ({
  type: constants.GET_ERROR,
  payload,
})

const getDocs = (payload: number): EnquiryAction => ({
  type: constants.GET_DOCS,
  payload,
})

const getDocsSuccess = (payload: Array<DocumentDTO>): EnquiryAction => ({
  type: constants.GET_DOCS_SUCCESS,
  payload,
})

const getDocsError = (payload: string): EnquiryAction => ({
  type: constants.GET_DOCS_ERROR,
  payload,
})
const updateDocs = (payload: number): EnquiryAction => ({
  type: constants.UPDATE_DOCS,
  payload,
})

const putDocType = (item: FileItem, docType: string): EnquiryAction => ({
  type: constants.PUT_DOC_TYPE,
  payload: {
    item: item,
    docType: docType,
  },
})

const putDocTypeSuccess = (payload: DocumentDTO): EnquiryAction => ({
  type: constants.PUT_DOC_TYPE_SUCCESS,
  payload,
})

const putDoc = (payload: FileItem): EnquiryAction => ({
  type: constants.PUT_DOC,
  payload,
})

const putDocSuccess = (payload: DocumentDTO): EnquiryAction => ({
  type: constants.PUT_DOC_SUCCESS,
  payload,
})

const putDocError = (payload: string): EnquiryAction => ({
  type: constants.PUT_DOC_ERROR,
  payload,
})

const getDoc = (payload: DocumentDTO): EnquiryAction => ({
  type: constants.GET_DOC,
  payload,
})

const getStdTemplate = (): EnquiryAction => ({
  type: constants.GET_STD_TEMPLATE,
  payload: null,
})

const getDocSuccess = (payload: DocData): EnquiryAction => ({
  type: constants.GET_DOC_SUCCESS,
  payload,
})

const getDocError = (payload: string): EnquiryAction => ({
  type: constants.GET_DOC_ERROR,
  payload,
})

const deleteDoc = (payload: number): EnquiryAction => ({
  type: constants.DELETE_DOC,
  payload,
})

const deleteDocSuccess = (payload: number): EnquiryAction => ({
  type: constants.DELETE_DOC_SUCCESS,
  payload,
})

const deleteDocError = (payload: string): EnquiryAction => ({
  type: constants.DELETE_DOC_ERROR,
  payload,
})

const resetNotifications = (payload: number): EnquiryAction => ({
  type: constants.RESET_NOTIFICATIONS,
  payload,
})

export default {
  init,
  initSuccess,
  initError,
  get,
  getSuccess,
  getError,
  getDocs,
  getDocsSuccess,
  getDocsError,
  updateDocs,
  putDoc,
  putDocSuccess,
  putDocType,
  putDocTypeSuccess,
  putDocError,
  getDoc,
  getStdTemplate,
  getDocSuccess,
  getDocError,
  deleteDoc,
  deleteDocSuccess,
  deleteDocError,
  resetNotifications,
}
