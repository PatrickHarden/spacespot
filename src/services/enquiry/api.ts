import axios, { AxiosResponse } from 'axios'
import CONFIG from 'config'
import { EnquiryResponse, DocumentDTO, EnquiryRequest } from './types'
import { FileItem } from 'services/global/types'
import { dataURItoBlob } from 'services/global/util'

export const initEnquiry = async (
  token: string,
  request: EnquiryRequest,
): Promise<EnquiryResponse> => {
  const url = request.loggedIn
    ? `${CONFIG.API_HOST}/enquiries`
    : `${CONFIG.API_HOST}/enquiries/v2`
  const config = {
    headers: {
      Authorization: request.loggedIn
        ? `Bearer ${token}`
        : 'U1BBQ0VTUE9UU1VCTUlURU5RVUlSWQ==',
    },
  }
  delete request.loggedIn // remove client-side specific property before going to API
  const resp: AxiosResponse<EnquiryResponse> = await axios.post(
    url,
    request,
    config,
  )
  return resp.data
}

export const getEnquiry = async (
  enquiryId: string,
  token: string,
): Promise<EnquiryResponse> => {
  const url = `${CONFIG.API_HOST}/enquiries/${enquiryId}`
  const config = { headers: { Authorization: `Bearer ${token}` } }
  const resp: AxiosResponse<EnquiryResponse> = await axios.get(url, config)
  return resp.data
}

export const putDocument = async (
  doc: FileItem,
  enquiryId: number,
  token: string,
): Promise<DocumentDTO> => {
  const url = `${CONFIG.API_HOST}/enquiries/${enquiryId}/documents`
  const formData = new FormData()
  formData.append('file', dataURItoBlob(doc.uri), doc.name)
  const resp: AxiosResponse<DocumentDTO> = await axios.post(url, formData, {
    headers: {
      'Content-Type': 'multipart/form-data',
      Authorization: `Bearer ${token}`,
    },
  })
  return resp.data
}

export const putDocumentType = async (
  doc: FileItem,
  enquiryId: number,
  token: string,
  docType: string,
): Promise<DocumentDTO> => {
  const url = `${CONFIG.API_HOST}/enquiries/${enquiryId}/documents`
  const formData = new FormData()
  formData.append('documentType', docType)
  formData.append('file', dataURItoBlob(doc.uri), doc.name)
  const resp: AxiosResponse<DocumentDTO> = await axios.post(url, formData, {
    headers: {
      'Content-Type': 'multipart/form-data',
      Authorization: `Bearer ${token}`,
    },
  })
  return resp.data
}

export const getStandardTemplate = async (token: string): Promise<any> => {
  const url = `${CONFIG.API_HOST}/leasedocuments/standardtemplate/download`
  const resp: AxiosResponse<any> = await axios.get(url, {
    responseType: 'blob',
    headers: {
      Authorization: `Bearer ${token}`,
      culture: 'default',
    },
  })
  return resp.data
}

export const getDocument = async (
  docId: number,
  enquiryId: number,
  token: string,
): Promise<any> => {
  const url = `${CONFIG.API_HOST}/enquiries/${enquiryId}/documents/${docId}`
  const resp: AxiosResponse<any> = await axios.get(url, {
    responseType: 'blob',
    headers: {
      Authorization: `Bearer ${token}`,
    },
  })
  return resp.data
}

export const deleteDocument = async (
  docId: number,
  enquiryId: number,
  token: string,
): Promise<any> => {
  const url = `${CONFIG.API_HOST}/enquiries/${enquiryId}/documents/${docId}`
  const resp: AxiosResponse<any> = await axios.delete(url, {
    headers: {
      Authorization: `Bearer ${token}`,
    },
  })
  return resp.data
}

export const getDocuments = async (
  enquiryId: number,
  token: string,
): Promise<Array<DocumentDTO>> => {
  const url = `${CONFIG.API_HOST}/enquiries/${enquiryId}/documents`
  const resp: AxiosResponse<Array<DocumentDTO>> = await axios.get(url, {
    headers: {
      'Content-Type': 'multipart/form-data',
      Authorization: `Bearer ${token}`,
    },
  })
  return resp.data
}

export const acceptEnquiry = async (
  enquiryId: string,
  token: string,
): Promise<EnquiryResponse> => {
  const url = `${CONFIG.API_HOST}/enquiries/${enquiryId}/accept`
  const config = { headers: { Authorization: `Bearer ${token}` } }
  const resp: AxiosResponse<EnquiryResponse> = await axios.post(url, config)
  return resp.data
}

export const resetNotifications = async (
  enquiryId: number,
  token: string,
): Promise<EnquiryResponse> => {
  const url = `${CONFIG.API_HOST}/enquiries/${enquiryId}/notifications`
  const config = { headers: { Authorization: `Bearer ${token}` } }
  const resp: AxiosResponse<EnquiryResponse> = await axios.put(url, config)
  return resp.data
}
