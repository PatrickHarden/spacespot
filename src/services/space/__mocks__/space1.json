{
  "Common.ActualAddress": {
    "Common.PostalAddresses": [
      {
        "Common.Language": "es-ES",
        "Common.Line1": "Bryggegata 29, 0250 Oslo, Noruega",
        "Common.Locallity": "Oslo"
      }
    ],
    "Common.Country": "NO",
    "Common.PostCode": "0250"
  },
  "Common.Brochures": [],
  "Common.BuildingOperatorID": "920d0a43-7322-43b1-a22c-01437939c880",
  "Common.ContactGroup": { "Common.GroupType": "AreaOffice" },
  "Common.Coordinate": { "lat": 59.91078, "lon": 10.726055 },
  "Common.FloorPlans": [
    {
      "Common.ImageCaption": "fplan1.jpg",
      "Common.ImageResources": [
        {
          "Common.Image.Height": 197,
          "Common.Resource.Uri": "/resources/fileassets/SS_SPACE_648/6862dd87/fplan1.jpg",
          "Common.Image.Width": 291
        }
      ]
    }
  ],
  "Common.FloorsAndUnits": [
    {
      "Common.SubdivisionName": [
        { "Common.CultureCode": "es-ES", "Common.Text": "HotDesk" }
      ],
      "Common.Areas": [
        {
          "Common.Units": "pp",
          "Common.Area": 1,
          "Common.MinArea": 1,
          "Common.MaxArea": 50
        }
      ],
      "Common.Unit.Status": "Available",
      "Common.Unit.Use": "HotDesk",
      "Common.AvailableFrom": "2020-02-19",
      "Common.Availability": {
        "Common.AvailabilityKind": "AvailableFromKnownDate",
        "Common.AvailabilityDate": "2020-02-19",
        "Common.MinLeaseTerm": { "Common.TermDuration": 0 }
      },
      "Common.Charges": [
        {
          "Common.ChargeKind": "Rent",
          "Common.CurrencyCode": "NOK",
          "Common.Interval": "Daily",
          "Common.Amount": 0.0,
          "Common.AmountKind": "Value",
          "Common.PerUnit": "unit",
          "Common.TaxModifer": "PlusVAT",
          "Common.OnApplication": false,
          "Common.ChargeModifer": "PriceIsNegotiable"
        }
      ]
    },
    {
      "Common.SubdivisionName": [
        { "Common.CultureCode": "es-ES", "Common.Text": "FixedDesk" }
      ],
      "Common.Areas": [
        {
          "Common.Units": "pp",
          "Common.Area": 1,
          "Common.MinArea": 1,
          "Common.MaxArea": 50
        }
      ],
      "Common.Unit.Status": "Available",
      "Common.Unit.Use": "FixedDesk",
      "Common.AvailableFrom": "2020-02-19",
      "Common.Availability": {
        "Common.AvailabilityKind": "AvailableFromKnownDate",
        "Common.AvailabilityDate": "2020-02-19",
        "Common.MinLeaseTerm": { "Common.TermDuration": 0 }
      },
      "Common.Charges": [
        {
          "Common.ChargeKind": "Rent",
          "Common.CurrencyCode": "NOK",
          "Common.Interval": "Daily",
          "Common.Amount": 0.0,
          "Common.AmountKind": "Value",
          "Common.PerUnit": "unit",
          "Common.TaxModifer": "PlusVAT",
          "Common.OnApplication": false,
          "Common.ChargeModifer": "PriceIsNegotiable"
        }
      ]
    },
    {
      "Common.SubdivisionName": [
        { "Common.CultureCode": "es-ES", "Common.Text": "ServicedOffice" }
      ],
      "Common.Areas": [
        {
          "Common.Units": "pp",
          "Common.Area": 1,
          "Common.MinArea": 1,
          "Common.MaxArea": 50
        }
      ],
      "Common.Unit.Status": "Available",
      "Common.Unit.Use": "ServicedOffice",
      "Common.AvailableFrom": "2020-02-19",
      "Common.Identifier": "1",
      "Common.Availability": {
        "Common.AvailabilityKind": "AvailableFromKnownDate",
        "Common.AvailabilityDate": "2020-02-19",
        "Common.MinLeaseTerm": { "Common.TermDuration": 12 }
      },
      "Common.Charges": [
        {
          "Common.ChargeKind": "Rent",
          "Common.CurrencyCode": "NOK",
          "Common.Interval": "Daily",
          "Common.Amount": 2.0,
          "Common.AmountKind": "Value",
          "Common.PerUnit": "unit",
          "Common.TaxModifer": "PlusVAT",
          "Common.OnApplication": false,
          "Common.ChargeModifer": "PriceIsNegotiable"
        }
      ]
    }
  ],
  "Common.Highlights": [
    {
      "Common.Highlight": [
        {
          "Common.CultureCode": "es-ES",
          "Common.Text": "Bryggegata 29 1 2020-02-18T17:24:32.497Z"
        }
      ]
    }
  ],
  "Common.HomeSite": "no-comm",
  "Common.IsParent": false,
  "Common.LastUpdated": "2020-02-18T17:25:08.8918759Z",
  "Common.LongDescription": [
    {
      "Common.CultureCode": "es-ES",
      "Common.Text": "Bryggegata 29 1 2020-02-18T17:24:32.497Z"
    }
  ],
  "Common.ParentProperty": "SS_BUILDING_646",
  "Common.Photos": [
    {
      "Common.ImageCaption": "photo1.jpg",
      "Common.ImageResources": [
        {
          "Common.Breakpoint": "original",
          "Common.Image.Height": 180,
          "Common.Image.Width": 223,
          "Common.Resource.Uri": "/resources/fileassets/SS_SPACE_648/b904b44f/photo1.jpg",
          "Source.Uri": "https://uatlistingssearchcbreeun.blob.core.windows.net/fileassets/SS_SPACE_648/b904b44f/photo1.jpg"
        },
        {
          "Common.Breakpoint": "small",
          "Common.Image.Height": 516,
          "Common.Image.Width": 640,
          "Common.Resource.Uri": "/resources/fileassets/SS_SPACE_648/b904b44f/photo1_Photo_1_small.jpg",
          "Source.Uri": "https://uatlistingssearchcbreeun.blob.core.windows.net/fileassets/SS_SPACE_648/b904b44f/photo1.jpg"
        },
        {
          "Common.Breakpoint": "medium",
          "Common.Image.Height": 1033,
          "Common.Image.Width": 1280,
          "Common.Resource.Uri": "/resources/fileassets/SS_SPACE_648/b904b44f/photo1_Photo_1_medium.jpg",
          "Source.Uri": "https://uatlistingssearchcbreeun.blob.core.windows.net/fileassets/SS_SPACE_648/b904b44f/photo1.jpg"
        },
        {
          "Common.Breakpoint": "large",
          "Common.Image.Height": 1291,
          "Common.Image.Width": 1600,
          "Common.Resource.Uri": "/resources/fileassets/SS_SPACE_648/b904b44f/photo1_Photo_1_large.jpg",
          "Source.Uri": "https://uatlistingssearchcbreeun.blob.core.windows.net/fileassets/SS_SPACE_648/b904b44f/photo1.jpg"
        }
      ]
    }
  ],
  "Common.PrimaryKey": "SS_SPACE_648",
  "Common.SourceLastupdated": "2020-02-18T17:25:08.680758Z",
  "Common.Strapline": [
    {
      "Common.CultureCode": "es-ES",
      "Common.Text": "Bryggegata 29 1 2020-02-18T17:24:32.497Z"
    }
  ],
  "Common.UsageType": "FlexOffice",
  "Common.Website": "https://www.spacespot.com",
  "Dynamic.HasHotDesks": false,
  "Dynamic.HasFixedDesks": false,
  "Dynamic.HasServicedOffices": false,
  "Dynamic.NewHome": false,
  "Dynamic.UnderOffer": false,
  "Dynamic.LetUnderOffer": false,
  "Dynamic.Furnished": "unfurnished",
  "Dynamic.HasSwimmingPool": false,
  "Dynamic.HasGym": false,
  "Dynamic.HasDoorkeeper": false,
  "Dynamic.IsPenthouse": false,
  "Dynamic.HasBalcony": false,
  "Dynamic.HasParking": false,
  "Dynamic.HasLift": false,
  "Dynamic.HasGarden": false,
  "Dynamic.DevelopmentOpportunity": false,
  "Dynamic.Has24HourAccess": false,
  "Dynamic.HasBreakoutSpace": false,
  "Dynamic.HasRoofTerrace": false,
  "Dynamic.HasWheelchairAccess": false,
  "Dynamic.HasTeaCoffee": false,
  "Dynamic.HasFreeMeetingRooms": false,
  "Dynamic.HasCleaning": false,
  "Dynamic.HasShowers": false,
  "Dynamic.HasReception": false,
  "Dynamic.HasBicycleRack": false,
  "Dynamic.HasTerrace": false,
  "Dynamic.HasAccessControl": false,
  "Dynamic.HasFloorsRaised": false,
  "Dynamic.HasStorageArchives": false,
  "Dynamic.HasWheelchairAccessToilets": false,
  "Dynamic.HasBarCafeteria": false,
  "Dynamic.HasBusinessLounge": false,
  "Dynamic.HasCoworkingArea": false,
  "Dynamic.HasGenerator": false,
  "Dynamic.HasElevators": false,
  "Dynamic.HasDayCare": false,
  "Dynamic.HasSecurity-24hours/7days": false,
  "Dynamic.HasFoodService-Cafeteria": false,
  "Dynamic.LettingCharge": null,
  "Dynamic.FlexCharge": null,
  "Dynamic.NetCharge": null,
  "Dynamic.SalePrice": null,
  "Dynamic.MinArea": "0",
  "Dynamic.TotalArea": null,
  "Dynamic.PrimaryImage": {
    "Common.ImageCaption": "photo1.jpg",
    "Common.ImageResources": [
      {
        "Common.Breakpoint": "original",
        "Common.Image.Height": 180,
        "Common.Image.Width": 223,
        "Common.Resource.Uri": "/resources/fileassets/SS_SPACE_648/b904b44f/photo1.jpg",
        "Source.Uri": "https://uatlistingssearchcbreeun.blob.core.windows.net/fileassets/SS_SPACE_648/b904b44f/photo1.jpg"
      },
      {
        "Common.Breakpoint": "small",
        "Common.Image.Height": 516,
        "Common.Image.Width": 640,
        "Common.Resource.Uri": "/resources/fileassets/SS_SPACE_648/b904b44f/photo1_Photo_1_small.jpg",
        "Source.Uri": "https://uatlistingssearchcbreeun.blob.core.windows.net/fileassets/SS_SPACE_648/b904b44f/photo1.jpg"
      },
      {
        "Common.Breakpoint": "medium",
        "Common.Image.Height": 1033,
        "Common.Image.Width": 1280,
        "Common.Resource.Uri": "/resources/fileassets/SS_SPACE_648/b904b44f/photo1_Photo_1_medium.jpg",
        "Source.Uri": "https://uatlistingssearchcbreeun.blob.core.windows.net/fileassets/SS_SPACE_648/b904b44f/photo1.jpg"
      },
      {
        "Common.Breakpoint": "large",
        "Common.Image.Height": 1291,
        "Common.Image.Width": 1600,
        "Common.Resource.Uri": "/resources/fileassets/SS_SPACE_648/b904b44f/photo1_Photo_1_large.jpg",
        "Source.Uri": "https://uatlistingssearchcbreeun.blob.core.windows.net/fileassets/SS_SPACE_648/b904b44f/photo1.jpg"
      }
    ]
  },
  "Dynamic.MaxArea": null,
  "Dynamic.MinDesks": "1",
  "Dynamic.TotalDesks": null,
  "Common.LastProcessed": "2020-02-18T17:27:14.4413615Z"
}
