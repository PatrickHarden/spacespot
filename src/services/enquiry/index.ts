import actions from './actions'
import constants from './constants'
import reducer from './reducer'
import saga from './saga'

export * from './types'
export { actions, constants, reducer, saga }
