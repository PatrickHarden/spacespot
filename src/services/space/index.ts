import actions from './actions'
import constants from './constants'
import reducer from './reducer'
import saga from './saga'
import selectors from './selectors'

export * from './types'
export { actions, constants, reducer, saga, selectors }
