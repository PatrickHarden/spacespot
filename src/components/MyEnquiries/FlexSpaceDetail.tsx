import React from 'react'
import { useIntl } from 'react-intl'
import { get } from 'lodash'

import { Space } from 'services/space/types'
import { getCountry, getFlexAvailability } from 'services/space/selectors'
import SpaceDetail from './SpaceDetail'
import { getRegionCurrencyCode } from 'services/global/region'
import { EnquiryType } from 'services/enquiry/types'
import Spinner from 'components/icons/Spinner'
import { AvailabilityFlex } from 'services/onboarding/types'

const getServiceOffice = (
  servicedOffices: Array<AvailabilityFlex>,
  flexIdentifier: number,
) => {
  let result = {}
  servicedOffices.forEach(servicedOffice => {
    if (servicedOffice.id === flexIdentifier.toString()) {
      result = servicedOffice
    }
  })

  return result
}

const FlexSpaceDetail = (props: {
  space: Space
  type?: EnquiryType
  flexIdentifier?: number
}) => {
  const { space, type, flexIdentifier } = props
  const { formatMessage } = useIntl()
  const t = (s: string) => formatMessage({ id: s })

  const currencyCode = getRegionCurrencyCode(getCountry(space))

  if (!space) {
    return <Spinner />
  }
  switch (type) {
    case EnquiryType.HotDesk: {
      const hotDesks = getFlexAvailability(space, 'HotDesk') as AvailabilityFlex
      return (
        <SpaceDetail
          quantityLabel={t('YOUR_ENQUIRIES_HOT_DESKS')}
          quantityValue={get(hotDesks, 'desks', 0).toString()}
          availability={get(hotDesks, 'availableFrom')}
          rent={get(hotDesks, 'price')}
          currencyCode={currencyCode}
        />
      )
    }
    case EnquiryType.FixedDesk: {
      const fixedDesks = getFlexAvailability(
        space,
        'FixedDesk',
      ) as AvailabilityFlex
      return (
        <SpaceDetail
          quantityLabel={t('YOUR_ENQUIRIES_FIXED_DESKS')}
          quantityValue={get(fixedDesks, 'desks', 0).toString()}
          availability={get(fixedDesks, 'availableFrom')}
          rent={get(fixedDesks, 'price')}
          currencyCode={currencyCode}
        />
      )
    }
    case EnquiryType.ServicedOffice:
    default: {
      const servicedOffices = getFlexAvailability(
        space,
        'ServicedOffice',
      ) as Array<AvailabilityFlex>
      const serviceOffice = getServiceOffice(
        servicedOffices,
        flexIdentifier as number,
      )
      return (
        <SpaceDetail
          quantityLabel={t('YOUR_ENQUIRIES_SERVICED_OFFICE')}
          quantityValue={`${get(serviceOffice, 'desks', 0).toString()} ${t(
            'DETAILS_SERVICED_OFFICES_PEOPLE',
          )}`}
          availability={get(serviceOffice, 'availableFrom')}
          rent={get(serviceOffice, 'price')}
          currencyCode={currencyCode}
        />
      )
    }
  }
}

export default FlexSpaceDetail
