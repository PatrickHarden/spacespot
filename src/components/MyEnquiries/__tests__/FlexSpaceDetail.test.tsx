import React from 'react'
import FlexSpaceDetail from '../FlexSpaceDetail'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import MockDate from 'mockdate'
import space1 from 'services/space/__mocks__/space1.json'
import { EnquiryType } from 'services/enquiry'
import { Space } from 'services/space'

jest.mock('react-intl', () => ({
  useIntl: () => ({ formatMessage: (s: any) => s.id }),
  createIntl: jest.fn(),
  createIntlCache: jest.fn(),
}))

it('renders without crashing each type of space', async () => {
  MockDate.set(1434319925275)
  const wrapperUndefined = shallow(
    <FlexSpaceDetail
      space={(undefined as unknown) as Space}
      type={EnquiryType.HotDesk}
    />,
  )
  expect(shallowToJson(wrapperUndefined)).toMatchSnapshot()

  const wrapperHotDesk = shallow(
    <FlexSpaceDetail
      space={(space1 as unknown) as Space}
      type={EnquiryType.HotDesk}
    />,
  )
  expect(shallowToJson(wrapperHotDesk)).toMatchSnapshot()

  const wrapperFixedDesk = shallow(
    <FlexSpaceDetail
      space={(space1 as unknown) as Space}
      type={EnquiryType.FixedDesk}
    />,
  )
  expect(shallowToJson(wrapperFixedDesk)).toMatchSnapshot()

  const wrapperServicedOffice = shallow(
    <FlexSpaceDetail
      space={(space1 as unknown) as Space}
      type={EnquiryType.ServicedOffice}
    />,
  )
  expect(shallowToJson(wrapperServicedOffice)).toMatchSnapshot()
  MockDate.reset()
})
