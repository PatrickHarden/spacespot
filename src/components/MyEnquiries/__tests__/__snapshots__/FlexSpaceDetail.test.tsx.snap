// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing each type of space 1`] = `<Styled(Component) />`;

exports[`renders without crashing each type of space 2`] = `
<SpaceDetail
  availability={2020-02-19T00:00:00.000Z}
  currencyCode="NOK"
  quantityLabel="YOUR_ENQUIRIES_HOT_DESKS"
  quantityValue="50"
  rent={0}
/>
`;

exports[`renders without crashing each type of space 3`] = `
<SpaceDetail
  availability={2020-02-19T00:00:00.000Z}
  currencyCode="NOK"
  quantityLabel="YOUR_ENQUIRIES_FIXED_DESKS"
  quantityValue="50"
  rent={0}
/>
`;

exports[`renders without crashing each type of space 4`] = `
<SpaceDetail
  availability={2020-02-19T00:00:00.000Z}
  currencyCode="NOK"
  quantityLabel="YOUR_ENQUIRIES_SERVICED_OFFICE"
  quantityValue="50 DETAILS_SERVICED_OFFICES_PEOPLE"
  rent={2}
/>
`;
